### PR TITLE
feat: added target_init conversion support

### PR DESCRIPTION
**Pull request by Mac-tf2**
Added support for target_init conversion and modified weapon spawning code to check for a targetname before proceeding with conversion.

### CODE REVIEW

File: BSPConversionLib/Source/EntityConverter.cs [hunk 1]
```diff
@@ -244,11 +255,61 @@ private void ConvertTriggerMultiple(Entity trigger)
...
+						ConvertInitTrigger(trigger, target);
+						break;
 				}
 			}
 
 			trigger["spawnflags"] = "1";
 		}
+		private void ConvertInitTrigger(Entity trigger, Entity targetInit)
+		{
+			var Spawnflags = (TargetInitFlags)targetInit.Spawnflags;
```
Comment (Panzerhandschuh) [inline BSPConversionLib/Source/EntityConverter.cs:268]: Use camelCase when naming local variables (should be named `spawnflags`)

File: BSPConversionLib/Source/EntityConverter.cs [hunk 2]
```diff
@@ -340,7 +401,7 @@ private void GiveWeaponOnStartTouch(Entity trigger, Entity weaponEnt)
 				target = "!activator",
 				action = "GiveDFWeapon",
 				param = weaponIndex.ToString(),
-				delay = 0f,
+				delay = 0.01f, //hack to make sure that the weapon removal applies before weapon give
```
Comment (Panzerhandschuh) [inline BSPConversionLib/Source/EntityConverter.cs:404]: You can leave this hack for now, but we should definitely find a better way of handling triggers that strip and give weapons simultaneously. 

Perhaps we just need to avoid calling RemoveWeaponOnStartTouch for any weapons referenced by the target_give?

File: BSPConversionLib/Source/EntityConverter.cs [hunk 3]
```diff
@@ -80,17 +91,17 @@ public void Convert()
 					case "target_give":
 						ignoreEntity = true;
 						break;
-					default:
-						{
-							if (entity.ClassName.StartsWith("weapon_"))
+						default:
```
Comment (Panzerhandschuh) [inline BSPConversionLib/Source/EntityConverter.cs:94]: The tab spacing is off here.

File: BSPConversionLib/Source/EntityConverter.cs [hunk 4]
```diff
@@ -80,17 +91,17 @@ public void Convert()
...
+							{
+							if ((entity.ClassName.StartsWith("weapon_")) && (entity.Name == ""))
```
Comment (Panzerhandschuh) [inline BSPConversionLib/Source/EntityConverter.cs:96]: Why are you checking if the targetname is blank? Is this meant to prevent weapons from getting converted before converting the give triggers? If so, you should revert this change and we should implement a proper fix instead.

Comment (Panzerhandschuh) [reply BSPConversionLib/Source/EntityConverter.cs:96]: I should also note that this will break maps with weapons that have a targetname assigned but no linked target_give.

Comment (Mac-tf2) [reply BSPConversionLib/Source/EntityConverter.cs:96]: do we not want to only convert weapon_ entities that aren't targets for this part?

File: BSPConversionLib/Source/EntityConverter.cs (hunk 2 above)
Comment (Mac-tf2) [reply BSPConversionLib/Source/EntityConverter.cs:404]: It's done this way in q3 because it resets ammo when the weapon is stripped away otherwise you can hit the trigger multiple times to add infinite ammo. I'm not sure how our ammo system works to be honest, so maybe that's not an issue in mmod.

**Comment by Panzerhandschuh** [comment]
You should link this PR to issue #16
